astropy/io/fits/fitsrec.py

```diff
@@ -478,9 +478,22 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
         return data
 
     def __repr__(self):
-        # Force use of the normal ndarray repr (rather than the new
-        # one added for recarray in Numpy 1.10) for backwards compat
-        return np.ndarray.__repr__(self)
+        repr_dtype = self.dtype
+        # FIXME: remove numpy.record in the dtype repr, as recarray?
+        # if repr_dtype.type is np.record:
+        #     repr_dtype = np.dtype((np.void, repr_dtype))
```

Comment:
`recarray`'s repr is removing the `numpy.record` in dtype's repr, would be nicer I think but this would break tests relying on repr we can maybe do this in a non-bugfix release.